### PR TITLE
Consistent naming scheme

### PR DESCRIPTION
**Pull request by nhudson**
## Description

Change Kubernetes object names to be more consistent

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Comment by nhudson** [comment]
Issue: #1458

File: deploy/helm-chart/templates/config.yaml [hunk 1]
```diff
@@ -4,17 +4,17 @@ metadata:
   name: {{ include "promscale.fullname" . }}
   namespace: {{ template "promscale.namespace" . }}
   labels:
-    app: {{ template "promscale.fullname" . }}
+    app: {{ include "promscale.fullname" . }}
     chart: {{ template "promscale.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
-    app.kubernetes.io/name: "promscale-connector"
+    app.kubernetes.io/name: "promscale"
```
Comment (paulfantom) [inline deploy/helm-chart/templates/config.yaml:11]:
```suggestion
    app.kubernetes.io/name: { include "promscale.fullname" . }}
```

Just an idea that came to my mind. What about having this particular label be the same as `app` label?

Comment (nhudson) [reply deploy/helm-chart/templates/config.yaml:11]: I thought about this yesterday as well.  I will make the change.

Comment (nhudson) [reply deploy/helm-chart/templates/config.yaml]: I just refactored how the labels are added to each template.  It should make it easier to manage in the future.

**Comment by nhudson** [comment]
I am not entirely sure that using the name of the Helm deployment is what we want.  Normally I would expect  the `app.kubernetes.io/*` labels should be mostly static and take values from either some static string or from the `Chart.yaml`.  Normally it would reflect the name of the helm chart being installed.  I am up for some discussion about it though.

File: deploy/helm-chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -33,4 +33,22 @@ Allow the release namespace to be overridden
...
+chart: {{ template "promscale.chart" . }}
+release: {{ .Release.Name }}
+heritage: {{ .Release.Service }}
+{{- end }}
+
+{{- define "promscale-helm.labels" -}}
+{{ include "promscale.labels" . }}
+app.kubernetes.io/name: {{ include "promscale.fullname" . | quote }}
+app.kubernetes.io/version: {{ .Chart.AppVersion }}
+app.kubernetes.io/part-of: {{ include "promscale.fullname" . | quote }}
```
Comment (paulfantom) [inline deploy/helm-chart/templates/_helpers.tpl:52]: We should remove this label as this helm chart is shipping a stack of applications.

As per https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels

**Review by paulfantom — APPROVED** [review]
LGTM, just one nit that we can take care of later.

**Comment by paulfantom** [comment]
@nhudson could you rebase to fix merge conflict?

**Review by onprem — APPROVED** [review]
LGTM!

**Comment by paulfantom** [comment]
Seems there are still some issues with rebase. Could you squash all the commits with manual changes into one commit and everything that is generated into another commit and then try to rebase it again?

**Pull request merged**